python/services/evaluationservice/dmod/evaluationservice/runner.py

```diff
@@ -1,60 +1,123 @@
 #!/usr/bin/env python3
+"""
+Listens to redis to identify messages indicating that an evaluation should be launched
+"""
+from __future__ import annotations
+
+import queue
 import typing
 import os
 import sys
 import multiprocessing
 import json
 import signal
+import dataclasses
+import threading
 
 from argparse import ArgumentParser
+from multiprocessing.pool import ApplyResult
+
+import redis
 
 import service
-import utilities
 import worker
+from utilities.common import ErrorCounter
+from utilities import streams
+
+EXCEPTION_LIMIT: typing.Final[int] = 10
 
 
-def signal_handler(signum, frame):
-    service.error("Received external signal. Now exiting.")
+def signal_handler(signum: int, frame):
```

Comment:
nit: May want to indicate that this function never returns:

```suggestion
def signal_handler(signum: int, frame) -> typing.Never:
```